python/paddle/distributed/launch/utils/process_context.py

```diff
@@ -36,7 +36,16 @@ def __init__(self,
 
     def _start(self):
         pre_fn = os.setsid if self._group else None
-        self._proc = subprocess.Popen(self._cmd,
+        # Need shell=True for IPU PopRun command
+        if "poprun" in self._cmd[0]:
```

Comment:
Consider add self._shell default False for process_context, add it to container too. 
Then handle it in controller with pod.

Let's keep process_context in low level and independent.